rocksdb_replicator/thrift/replicator.thrift

```diff
@@ -60,6 +60,7 @@ enum ErrorCode {
   OTHER = 0,
   SOURCE_NOT_FOUND = 1, # could not find the upstream db
   SOURCE_READ_ERROR = 2,
+  SOURCE_REMOVED = 3,
```

Comment:
where/when will we use this in the code?

Reply:
Since both the thrift file change and using the new field cannot be part of the same diff (will cause problems during rollout), I removed the part where this is used. This will be used [here](https://github.com/pinterest/rocksplicator/blob/master/rocksdb_replicator/replicated_db.cpp#L266)